### PR TITLE
[new page] e-commerce guide

### PR DESCRIPTION
**Pull request by sarah11918**
#### Description (required)

This adds a new Guide "E-commerce" to briefly introduce and mention integrating e-commerce options into an Astro project.

Goals:
- have some SEO/Algolia search hits when people wonder, "Can I do e-commerce with Astro?" (literal existance)
- present two different ways of approaching e-commerce in Astro to address different levels of need:
		- **Basic:** payment processing overlays (cart is basically handled for you; you add a script then some attributes to an HTML element to make it a "buy now button") which are great for static sites selling a small number of individual products.
		- **Advanced:** a full shopping cart experience (powerful and customizable) for those who want an online store.
- show Astro-friendly options at both levels, to give people somewhere to start if they know nothing about e-commerce.
		- In particular, we have both a full community Snipcart theme/design system option *and* a friendly helper community integration, both by active community members. People using either of these options are set up for success because we know they can receive support.

Non-Goals:
- complete tutorials. These are for people to *see what e-commerce in Astro looks like*: 
		- it can be as simple as adding a `<script>` then some custom classes/attributes to an HTML element.
		- If you don't need a "storefront", you don't need to build one to be able to sell things on your Astro site.
		- If you do want a more all-inclusive option, we show not only an example of basic usage, but also an Astro integration and a full Astro theme

### CODE REVIEW

**Review by VoxelMC — APPROVED** [review]
This looks awesome Sarah! I did a once over real quick, and I think it's really cool. Now I want to make an e-commerce site, even though I have nothing to sell!

I left one comment, I hope that it is useful :)

File: src/content/docs/en/guides/ecommerce.mdx [hunk 1]
```diff
@@ -0,0 +1,268 @@
+---
+title: E-commerce
+description: An introduction to adding e-commerce options to your Astro site
+i18nReady: true
+---
+import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'
+import ReadMore from '~/components/ReadMore.astro'
+
+Astro supports several e-commerce options, from checkout links to hosted payment pages to building an entire storefront using a payment service API.
```
[the remark this answers is not in the document]
Comment (sarah11918) [reply src/content/docs/en/guides/ecommerce.mdx:9]: Like it!

File: src/content/docs/en/guides/ecommerce.mdx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+1. Add the following two `<script>` tags to your page `head` or `body`:
+
+    ```html title="src/pages/my-product-page.astro"
+    <script src="https://cdn.paddle.com/paddle/v2/paddle.js"></script>
+    <script type="text/javascript">
+      Paddle.Setup({ 
+        token: '7d279f61a3499fed520f7cd8c08' // replace with a client-side token
+    });
+    </script>
```
Comment (TheOtterlord) [inline src/content/docs/en/guides/ecommerce.mdx:64]:
```suggestion
    <script type="text/javascript">
      Paddle.Setup({ 
        token: '7d279f61a3499fed520f7cd8c08' // replace with a client-side token
      });
    </script>
```

File: src/content/docs/en/guides/ecommerce.mdx [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+    ```html title="src/pages/my-product-page.astro"
+    <script src="https://app.lemonsqueezy.com/js/lemon.js" defer></script>
+    ```
+2. Create an anchor tag on the page linking to your product URL. Include the class `lemonsqueezy-button` to open a checkout overlay when clicked.
+
+    ```html title="src/pages/my-product-page.astro"
+    <a class="lemonsqueezy-button" href="https://demo.lemonsqueezy.com/checkout/...">
+        Buy Now
+    </a>
```
Comment (TheOtterlord) [inline src/content/docs/en/guides/ecommerce.mdx:35]:
```suggestion
    <a class="lemonsqueezy-button" href="https://demo.lemonsqueezy.com/checkout/...">
      Buy Now
    </a>
```

File: src/content/docs/en/guides/ecommerce.mdx [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+### Paddle
+
+[Paddle](https://www.paddle.com/) is a billing solution for digital products and services. It handles handle payments, taxes, and subscription management through an overlay or inline checkout. 
+
+[Paddle.js](https://developer.paddle.com/paddlejs/overview) is a lightweight JavaScript library that lets you build rich, integrated subscription billing experiences using Paddle.
+
+#### Basic Usage
+
+The following is an example of adding a Paddle "Buy Now" element to an Astro page. Clicking this link will open a checkout an allow the visitor to complete a single purchase.
```
Comment (TheOtterlord) [inline src/content/docs/en/guides/ecommerce.mdx:52]:
```suggestion
The following is an example of adding a Paddle "Buy Now" element to an Astro page. Clicking this link will open a checkout and allow the visitor to complete a single purchase.
```

File: src/content/docs/en/guides/ecommerce.mdx [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+
+### Lemon Squeezy
+
+[Lemon Squeezy](https://www.lemonsqueezy.com/) is an all-in-one platform for payments and subscriptions with multi-currency support, global tax compliance, PayPal integration and more. It allows you to create and manage digital products and services through your account dashboard and provides product URLs for the checkout process.
+
+The basic [Lemon.js JavaScript library](https://docs.lemonsqueezy.com/help/lemonjs/what-is-lemonjs) allows you to sell your Lemon Squeezy products with a checkout link.
+
+#### Basic Usage
+
+The following is an example of adding a Lemon Squeezy "Buy now" element to an Astro page. Clicking this link will open a checkout an allow the visitor to complete a single purchase.
```
Comment (TheOtterlord) [inline src/content/docs/en/guides/ecommerce.mdx:23]:
```suggestion
The following is an example of adding a Lemon Squeezy "Buy now" element to an Astro page. Clicking this link will open a checkout and allow the visitor to complete a single purchase.
```

File: src/content/docs/en/guides/ecommerce.mdx [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+
+    ```html title="src/pages/my-product-page.astro"
+    <button class="snipcart-checkout">Click here to checkout</button>
+    ```
+
+#### Snipcart JavaScript SDK
+
+The [Snipcart JavaScript SDK](https://docs.snipcart.com/v3/sdk/basics) lets you configure, customize and manage your Snipcart cart programmatically.
+
+This allows you to allows you to perform actions such as:
```
Comment (TheOtterlord) [inline src/content/docs/en/guides/ecommerce.mdx:177]:
```suggestion
This allows you to perform actions such as:
```

File: src/content/docs/en/guides/ecommerce.mdx [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+
+<ReadMore>See the [Snipcart documentation](https://docs.snipcart.com/v3/) for more more information about all the options to integrate Snipcart with your Astro Project.</ReadMore>
+
+#### `astro-snipcart`
+
+There are two `astro-snipcart` community packages that can simplify using Snipcart.
+
+- [`@lloydjatkinson/astro-snipcart` Astro template](https://astro-snipcart.vercel.app/): This Astro template includes an optional design system for a complete e-commerce solution out of the box. Learn more on its own extensive documentation site, including [the motivation behind building `astro-snipcart`](https://astro-snipcart.vercel.app/motivation) as providing a convenient, Astro-native way for you to interact with the Snipcart API.
+
+- [`@Adammatthiesen/astro-snipcart` integration](https://github.com/Adammatthiesen/astro-snipcart): This integration was heavily inspired by the `astro-snipcart` theme and provides Astro components (or Vue components) that you can add to your existing Astro project for creating products, controling the cart, and more. See the [full tutorial](https://matthiesen.xyz/blog/getting-started-with-my-astro-snipcart-addon) for more information.
```
Comment (TheOtterlord) [inline src/content/docs/en/guides/ecommerce.mdx:191]:
```suggestion
- [`@Adammatthiesen/astro-snipcart` integration](https://github.com/Adammatthiesen/astro-snipcart): This integration was heavily inspired by the `astro-snipcart` theme and provides Astro components (or Vue components) that you can add to your existing Astro project for creating products, controlling the cart, and more. See the [full tutorial](https://matthiesen.xyz/blog/getting-started-with-my-astro-snipcart-addon) for more information.
```

**Review by TheOtterlord — APPROVED** [review]
This looks amazing! I've left a few minor typo & formatting suggestions, but other than those, this feels good to go!

**Comment by lloydjatkinson** [comment]
LGTM

**Comment by kevinzunigacuellar** [comment]
LGTM!

**Review by Adammatthiesen — APPROVED** [review]
Looks amazing <3

**Comment by rgilsimoes** [comment]
Participated in the group review at Discord.

**Comment by sarah11918** [comment]
!coauthor

**Comment by sarah11918** [comment]
!coauthor

**Pull request merged**